h2/stream.py

```diff
@@ -216,23 +224,31 @@ def send_new_pushed_stream(self, previous_state):
         """
         Fires on the newly pushed stream, when pushed by the local peer.
 
-        No event here, but definitionally this peer must be a server.
+        Only an internal event here, but definitionally this peer must
+        be a server.
         """
         assert self.client is None
         self.client = False
         self.headers_received = True
-        return []
+
+        event = _PushPromiseSent()
+        event.stream_id = self.stream_id
+        return [event]
 
     def recv_new_pushed_stream(self, previous_state):
         """
         Fires on the newly pushed stream, when pushed by the remote peer.
 
-        No event here, but definitionally this peer must be a client.
+        Only an internal event here, but definitionally this peer must
+        be a client.
         """
         assert self.client is None
         self.client = True
         self.headers_sent = True
-        return []
+
+        event = _PushPromiseSent()
+        event.stream_id = self.stream_id
+        return [event]
```

Comment:
This doesn't seem right. When we receive a push promise we spin *two* state machine methods. Conceptually, one of them owns the pushed headers and the other one does not.

We don't need a `PushPromiseSent` event in this case, because we *receive* a `PushPromiseReceived` event.